### PR TITLE
chore: update conlinmarc/hdfs to v2.2.1. Fixes #7816

### LINKED ISSUE #7816: Update `colinmarc/hdfs` to  version 2.x
# Summary

The 2.x releases notes contains breaking changes: https://github.com/colinmarc/hdfs/releases/tag/v2.0.0

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by yashvardhan-kukreja**
Signed-off-by: Yashvardhan Kukreja <yash.kukreja.98@gmail.com>

Fixes #7816 

Updates:
- `colinmark/hdfs v1.1.4` to `colinmark/hdfs/v2 v2.2.1`
- `gopkg.in/jcmturner/gokrb5.v5 v5.3.0` to `github.com/jcmturner/gokrb5/v8 v8.4.2`

### CODE REVIEW

**Review by terrytangyuan — COMMENTED** [review]
Do you have HDFS to test out the changes?

File: go.mod [hunk 1]
```diff
@@ -77,7 +75,11 @@ require (
 	upper.io/db.v3 v3.6.3+incompatible
 )
 
-require github.com/go-git/go-git/v5 v5.3.0
+require (
```
Comment (blkperl) [inline go.mod:78]: I think this should be in the require block above instead of separate.

Comment (yashvardhan-kukreja) [reply go.mod:78]: Sure, I'll do so. The above go.mod related changes were the magic of go mod tidy :)

**Comment by sarabala1979** [comment]
Have you tested with HDFS env? what is HDFS version?

**Comment by yashvardhan-kukreja** [comment]
Hi @sarabala1979 @terrytangyuan , sorry I was pre-occupied a bit due to which I couldn't test against an existing HDFS setup.

I haven't played with the HDFS side of things associated argo-workflows so do you mind sharing template/guide/reference for hdfs<->argo-workflows setup which I can simulate locally and test this PR against ? :)

**Comment by blkperl** [comment]
@dtaniwaki as the original contributor of hdfs support would you be able to help manually test this PR?

File: workflow/artifacts/hdfs/hdfs.go [hunk 2]
```diff
@@ -7,8 +7,8 @@ import (
 	"path/filepath"
 
 	"github.com/argoproj/pkg/file"
-	"gopkg.in/jcmturner/gokrb5.v5/credentials"
-	"gopkg.in/jcmturner/gokrb5.v5/keytab"
+	credentialsv8 "github.com/jcmturner/gokrb5/v8/credentials"
+	keytabv8 "github.com/jcmturner/gokrb5/v8/keytab"
```
Comment (NikeNano) [inline workflow/artifacts/hdfs/hdfs.go:11]: nit: previously the version of the package where not part of the name, what is the advantage of adding it?

**Comment by dtaniwaki** [comment]
@blkperl Sorry for the late reply. I’m working in a different org and we don’t have HDFS right now. My current laptop doesn’t have it neither as I don’t use it now…

File: workflow/artifacts/hdfs/hdfs.go (hunk 2 above)
Comment (yashvardhan-kukreja) [reply workflow/artifacts/hdfs/hdfs.go:11]: No certain reason tbh, just kept it at that moment. 
I can happily get rid of the `v8`s in the naming :) 

PS: Apologies for the late reply, my github notifications clutter is a mess to deal with :P

**Pull request closed without merging**